--- setup.cfg.orig  2022-02-11 12:39:12.092113700 +0100
+++ setup.cfg       2022-02-11 12:40:16.905071900 +0100
@@ -7,8 +7,24 @@
 provides = pyhdf
 doc_files = pyhdf/doc/
 
+[metadata]
+license_file = LICENSE
+license_files =
+    LICENSE
+    LICENSE.libhdf
+    LICENSE.libjpeg
+    LICENSE.zlib
+
 [options]
-install_requires = numpy
+install_requires =
+    numpy >= 1.11, < 1.12; python_version == "2.6"
+    numpy >= 1.16, < 1.17; python_version == "2.7"
+    numpy >= 1.11, < 1.12; python_version == "3.2"
+    numpy >= 1.11, < 1.12; python_version == "3.3"
+    numpy >= 1.15, < 1.17; python_version == "3.4"
+    numpy >= 1.16, < 1.19; python_version == "3.5"
+    numpy >= 1.16, < 1.20; python_version == "3.6"
+    numpy >= 1.21, < 1.23; python_version >= "3.7"
 setup_requires = numpy
 
 [egg_info]
