--- a/numpy/core/src/multiarray/einsum.c.src.orig  2021-01-04 15:16:38.000000000 +0100
+++ b/numpy/core/src/multiarray/einsum.c.src       2022-01-06 00:05:22.177030533 +0100
@@ -2152,7 +2152,7 @@
         }
         /* A repeated label, find the original one and merge them. */
         else {
-#ifdef __GNUC__
+#if defined(__GNUC__) && __GNUC__ >= 5
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wuninitialized"
 #pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
@@ -2169,7 +2169,7 @@
                 return NULL;
             }
             new_strides[i] += stride;
-#ifdef __GNUC__
+#if defined(__GNUC__) && __GNUC__ >= 5
 #pragma GCC diagnostic pop
 #endif
         }