--- setupext.py.orig	2022-03-02 16:41:31.117000000 +0100
+++ setupext.py	2022-03-02 17:02:44.218727700 +0100
@@ -855,10 +855,25 @@
                                   'NPY_1_7_API_VERSION'))
 
     def get_setup_requires(self):
-        return ['numpy>=1.6']
+        import sys
+        if sys.version_info[:2] == (3, 2):
+            return [
+                'numpy >= 1.11, < 1.12'
+            ]
+        else:
+            return [
+                'numpy >= 1.11, < 1.12; python_version == "2.6"',
+                'numpy >= 1.16, < 1.17; python_version == "2.7"',
+                'numpy >= 1.11, < 1.12; python_version == "3.2"',
+                'numpy >= 1.11, < 1.12; python_version == "3.3"',
+                'numpy >= 1.15, < 1.17; python_version == "3.4"',
+                'numpy >= 1.16, < 1.19; python_version == "3.5"',
+                'numpy >= 1.16, < 1.20; python_version == "3.6"',
+                'numpy >= 1.21, < 1.23; python_version >= "3.7"',
+            ]
 
     def get_install_requires(self):
-        return ['numpy>=1.6']
+        return self.get_setup_requires()
 
 
 class LibAgg(SetupPackage):
@@ -1186,7 +1201,11 @@
         return "using cycler version %s" % cycler.__version__
 
     def get_install_requires(self):
-        return ['cycler']
+        import sys
+        if sys.version_info[:2] == (3, 2):
+            return ['cycler < 0.11']
+        else:
+            return ['cycler']
 
 
 class Dateutil(SetupPackage):
@@ -1275,6 +1294,11 @@
 
     def get_install_requires(self):
         versionstring = 'pyparsing>=1.5.6,!=2.0.4,!=2.1.2'
+        import sys
+        if sys.version_info[:2] == (2, 6):
+            versionstring += ",<2.4.1"
+        elif sys.version_info[:2] == (3, 2):
+            versionstring += ",<2.3.1"
         if self.is_ok():
             return [versionstring]
         else:
