--- setup.py.orig  2022-03-01 20:36:44.435000000 +0100
+++ setup.py       2022-03-01 20:40:35.606093000 +0100
@@ -212,6 +212,25 @@
     except:
         build_requires = ['numpy>=1.6.2']
 
+    # Overwrite original `build_requires` and `install_requires`.
+    if sys.version_info[:2] == (3, 2):
+        install_requires = [
+            'numpy >= 1.11, < 1.12',
+        ]
+    else:
+        install_requires = [
+            'numpy >= 1.11, < 1.12; python_version == "2.6"',
+            'numpy >= 1.16, < 1.17; python_version == "2.7"',
+            'numpy >= 1.11, < 1.12; python_version == "3.2"',
+            'numpy >= 1.11, < 1.12; python_version == "3.3"',
+            'numpy >= 1.15, < 1.17; python_version == "3.4"',
+            'numpy >= 1.16, < 1.19; python_version == "3.5"',
+            'numpy >= 1.16, < 1.20; python_version == "3.6"',
+            'numpy >= 1.21, < 1.23; python_version >= "3.7"',
+        ]
+    if build_requires:
+        build_requires = install_requires
+
     metadata = dict(
         name='scipy',
         maintainer="SciPy Developers",
@@ -226,7 +245,7 @@
         platforms=["Windows", "Linux", "Solaris", "Mac OS-X", "Unix"],
         test_suite='nose.collector',
         setup_requires=build_requires,
-        install_requires=build_requires,
+        install_requires=install_requires,
     )
 
     if len(sys.argv) >= 2 and ('--help' in sys.argv[1:] or
